core/staking_verifier.go

```diff
@@ -172,9 +174,12 @@ func VerifyAndDelegateFromMsg(
 				); err != nil {
 					return nil, nil, err
 				}
-				// Return remaining balance to be deducted for delegation
 				if delegateBalance.Cmp(big.NewInt(0)) < 0 {
-					return nil, nil, errInsufficientBalanceForStake // shouldn't really happen
+					return nil, nil, errNegativeAmount // shouldn't really happen
+				}
+				// Return remaining balance to be deducted for delegation
+				if !CanTransfer(stateDB, msg.DelegatorAddress, delegateBalance) {
+					return nil, nil, errInsufficientBalanceForStake
```

Comment:
would be nice , `errors.Wrapf(errInsufficientBalanceForStake, "had %v, tried to stake %v", balance, stakeWanted)`

Reply:
done